datastore/prometheus.go

```diff
@@ -176,51 +178,29 @@ func (p *PrometheusExporter) handlePacket(pkt Packet) {
 		"toPort": strconv.Itoa(int(pkt.ToPort)),
 	}
 
-	if pkt.FromType == PodSource {
+	// We only want to keep metrics between pods in the same project (app namespace)
+	if pkt.FromType == PodSource && pkt.ToType == PodDest {
 		fromPod, found := p.podCache.get(pkt.FromUID)
-		if found {
+		toPod, found2 := p.podCache.get(pkt.ToUID)
+
+		// Make sure that both pods exist in the cache, have a project label set, and have the same project and app names
+		if found && found2 && fromPod.(PodEvent).Labels[appNamespaceLabel] != "" && toPod.(PodEvent).Labels[appNamespaceLabel] != "" &&
```

Comment:
Do you need to check if `fromPod.(PodEvent).Labels[appLabel]` is not empty too?

Reply:
No, as there won't be a case where this is unset but appNamespaceLabel is set, unless there is a crazy bug in runtime. And even if that were the case, it is not a big deal, since we still know it is traffic at least within the same project.

Reply:
👍